core/src/test/java/com/linecorp/armeria/client/EndpointTest.java

```diff
@@ -255,6 +265,19 @@ void ipV6Parse() {
         assertThat(a.toUri("none+http").toString()).isEqualTo("none+http://[::1]:80");
     }
 
+    @Test
+    void domainSocket() {
+        final String sockAddr = "unix%3Afoo.sock";
+        final List<Function<String, Endpoint>> factoryFunctions =
+                ImmutableList.of(Endpoint::of, Endpoint::parse);
+
+        factoryFunctions.forEach(factoryFunc -> {
+            final Endpoint e = factoryFunc.apply(sockAddr);
+            assertThat(e.host()).isEqualTo(sockAddr);
+            assertThat(e.isDomainSocket()).isTrue();
+        });
```

Comment:
Should we instead throw an exception? 🤔

Reply:
Raising an exception sounds good.

Reply:
Hmm, I actually changed `Endpoint` to silently enforce the predefined IP address and port number on domain socket endpoints. The reasoning behind this behavior is that it's more consistent with the behavior of `withPort()`:

```java
var e = Endpoint.of(DomainSocketAddress.of(...));
assert e.withPort(300) == e;
```

Why does `withPort()` not raise an exception? It's basically because of:

- Backward compatibility: Existing code that calls `Endpoint.withPort()` doesn't break when they start using domain sockets.
- Simpler code: Users don't have to check `isDomainSocket()` every time they call `withPort()` or `withIpAddr()`.